### PR TITLE
Move DrawView to separate file

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
## Summary

Refactoring. Moves DrawView out of activity to a separate file. 

## Safety Assurance

- [x] If the PR is high risk, "High Risk" label is set
- [x] I have confidence that this PR will not introduce a regression for the reasons below
- [x] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
Build will fail as of now because of this https://github.com/dimagi/commcare-android/pull/2497/commits/a8c72416cc7c4632f8bf60489d4321207ff6ded4. 
This PR will be ready for  review after we merge `signature-required` branch into it.

File: app/res/layout/draw_layout.xml [hunk 1]
```diff
@@ -20,6 +20,7 @@
         <Button
             android:id="@+id/btnFinishDraw"
             android:layout_width="wrap_content"
+            android:enabled="false"
             android:layout_height="wrap_content"/>
 
         <LinearLayout
```
Comment (dimagimon) [inline app/res/layout/draw_layout.xml:26]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/res/layout/draw_layout.xml#L26"></span><code>LinearLayout-3</code> can overlap <code>@id/btnFinishDraw</code> if @id/btnFinishDraw, LinearLayout-3 grow due to localized text expansion</td>
    </tr>
  </tbody>
</table>

File: app/src/org/commcare/views/DrawView.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import android.view.View;
+import android.view.WindowManager;
+
+import org.commcare.dalvik.R;
+import org.commcare.utils.MediaUtil;
+import org.commcare.views.widgets.SignatureWidget;
+
+import java.io.File;
+
+public class DrawView extends View {
```
Comment (dimagimon) [inline app/src/org/commcare/views/DrawView.java:19]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/views/DrawView.java#L19"></span>Custom view <code>DrawView</code> is missing constructor used by tools: <code>(Context)</code> or <code>(Context,AttributeSet)</code> or <code>(Context,AttributeSet,int)</code></td>
    </tr>
  </tbody>
</table>

File: app/src/org/commcare/views/DrawView.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            mCurrentPath.lineTo(mX, mY);
+            // commit the path to our offscreen
+            mCanvas.drawPath(mCurrentPath, paint);
+        }
+        // kill this so we don't double draw
+        mCurrentPath.reset();
+    }
+
+    @Override
+    public boolean onTouchEvent(MotionEvent event) {
```
Comment (dimagimon) [inline app/src/org/commcare/views/DrawView.java:138]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/views/DrawView.java#L138"></span>Custom view <code>DrawView</code> overrides <code>onTouchEvent</code> but not <code>performClick</code></td>
    </tr>
  </tbody>
</table>

**Comment by dimagimon** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          8 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR does not contain any JIRA issue keys in the PR title or commit messages (e.g. KEY-123)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/res/layout/draw_layout.xml#L32">app/res/layout/draw_layout.xml#L32</a> - Buttons in button bars should be borderless; use <code>style="?android:attr/buttonBarButtonStyle"</code> (and <code>?android:attr/buttonBarStyle</code> on the parent)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/res/layout/draw_layout.xml#L37">app/res/layout/draw_layout.xml#L37</a> - Buttons in button bars should be borderless; use <code>style="?android:attr/buttonBarButtonStyle"</code> (and <code>?android:attr/buttonBarStyle</code> on the parent)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/activities/DrawActivity.java#L113">app/src/org/commcare/activities/DrawActivity.java#L113</a> - Using 'Log' instead of 'Timber'</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/activities/DrawActivity.java#L124">app/src/org/commcare/activities/DrawActivity.java#L124</a> - Using 'Log' instead of 'Timber'</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/activities/DrawActivity.java#L164">app/src/org/commcare/activities/DrawActivity.java#L164</a> - Avoid passing <code>null</code> as the view root (needed to resolve layout parameters on the inflated layout's root element)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/activities/DrawActivity.java#L238">app/src/org/commcare/activities/DrawActivity.java#L238</a> - Using 'Log' instead of 'Timber'</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/dimagi/commcare-android/blob/fd97cc14c2cf8a94c905831da7cd406d17d4705b/app/src/org/commcare/activities/DrawActivity.java#L262">app/src/org/commcare/activities/DrawActivity.java#L262</a> - Using 'Log' instead of 'Timber'</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by damagatchi** [comment]
[33% (0.0%) vs master 33%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462484/)

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**